### ISSUE TITLE
Cannot start multiple threads

### ISSUE THREAD

**Issue by AnyiHu** [shafferm/DRAM#125]
Hey @shafferm , it is an awesome annotation tool. However, I cannot start multiple threads when I tried to annotate MAGs (~1000) I got by using the DRAM. I have no idea on it, but didn't meet similar issue when using other software. Could you provide some suggestions? Thanks.

Right now, I splitted the dataset into several groups and annotated them seperately using DRAM.

My server has 4 processors, each of which contained 15 cores. So, a total of 60 cores is available.
![DRAM_running](https://user-images.githubusercontent.com/57850945/142721638-6cab934c-5e76-4f7d-b45e-29c152034a44.jpg)

**Comment by rmFlynn** [comment]
It looks like dram is doing a hmm search, which  is limited to 2 threads max other parts of the process will use all threads. If that is not the casey, then it is hard to know more without seeing your commands and knowing more about your environment. Let me know how you are running dram, what type of machine, if you are using slum or similar, and anything that may be relevant. Also post your command.

**Comment by strejcem** [comment]
Hi,
I believe what @AnyiHu is seeing is normal behavior i.e., 2 threads are being used per one MAG/bin. However I have different issue. DRAM in my case is running only 1 thread and only one MAG at a time. There must be some issue that on my system only one thread is detected. I guess the issue is connected to me running WSL2. Furthermore, I have a AMD Ryzen 5950x CPU, which might complicate things. Can anyone point me which part of the code is responsible for the parallelization over MAGs? So I can try to resolve this? I don't python very well.  
Any help would be appreciated. Thanks!

**Comment by rmFlynn** [comment]
Yes, that is correct at the moment. This is one more thing that we are hoping to change. Still more than one CPU should be used depending on your commands.

**Comment by JoseLopezArcondo** [comment]
Hello, I have I think the same issue as here, I run
DRAM.py annotate -i 'dram_genomes/*.fasta' -o dram_genomes_annotation --threads 20
But seems to be using only one thread and one MAG at a time.. this would be unfeasible given the amount of MAGs I have to annotate (ca. 7000).
![Screenshot from 2023-02-04 20-56-32](https://user-images.githubusercontent.com/107948044/216794602-29ed239f-c549-4e2f-a63c-f056e2f14601.png)
Am I missing something here?
Thanks, amazing tool
